src/ejabberd_gen_sm.erl

```diff
@@ -32,78 +35,76 @@
          get_sessions/4, create_session/5, update_session/5, delete_session/5, cleanup/2,
          total_count/1, unique_count/1]).
 
--ignore_xref([behaviour_info/1, cleanup/2]).
+-ignore_xref([cleanup/2, behaviour_info/1]).
 
--spec start(module(), list()) -> any().
-start(Mod, Opts) ->
-    Mod:start(Opts).
+-define(MAIN_MODULE, ejabberd_sm).
+-type host_type() :: mongooseim:host_type() | global.
```

Comment:
This is definition is repeated with many names in many many places, like gen_hook, acl, metrics... Shouldn't we by now just make it a global type defined in `mongooseim.erl` together with the original host_type(), like
```erl
-type host_type_or_global() :: mongooseim:host_type() | global.
```
And just use it everywhere?
No need for this PR to retroactively modify all places where this is already used, that would be a too big diff for this PR, but the definition can already be created and we'll update everything else later.

Reply:
Maybe not in this PR, I'd personally go with just using `mongooseim:host_type() | global` in type specs, but I think just renaming this `host_type()` to `host_type_or_global()` would be fine for now.